packages/bson-transpilers/symbols/php/templates.yaml

```diff
@@ -0,0 +1,1231 @@
+#                                           #
+#               Good to know:               #
+# lhs is left-hand-side of the expression   #
+# rhs is right-hand-side of the expression  #
+# All args are strings unless noted         #
+#    - arg? is boolean                      #
+#    - arg# is number                       #
+#                                           #
+#############################################
+Templates:
+    ########
+    # Misc #
+    ########
+
+    # Filter out regex flags that have translations or are unsupported.
+    RegexFlags: &RegexFlags
+        i: 'i'
+        m: 'm'
+        u: ''
+        y: ''
+        g: ''
+    BSONRegexFlags: &BSONRegexFlags
+        i: 'i'
+        m: 'm'
+        x: 'x'
+        s: ''
+        l: ''
+        u: ''
+
+    #############################################
+    #                  Syntax                   #
+    #                                           #
+    # Templates for language syntax expressions #
+    #                                           #
+    #############################################
+
+    DriverTemplate: &DriverTemplate !!js/function >
+        (spec) => {
+            const getKey = k => {
+                let translateKey = {
+                    project: 'projection',
+                }
+                return k in translateKey ? translateKey[k] : k
+            };
+            const options = spec.options;
+            const filter = spec.filter || {};
+            delete spec.options;
+            delete spec.filter;
+
+            comment = []
+                .concat('// Requires the MongoDB PHP Driver')
+                .concat('// https://www.mongodb.com/docs/drivers/php/')
+                .join('\n')
+            ;
+            const client = `$client = new Client('${options.uri}');`;
+            const collection = `$collection = $client->selectCollection('${options.database}', '${options.collection}');`;
+
+            if ('aggregation' in spec) {
+                return []
+                    .concat(comment)
+                    .concat('')
+                    .concat(client)
+                    .concat(collection)
+                    .concat(`$cursor = $collection->aggregate(${this.utils.toPHPArray(spec.aggregation)});`)
+                    .join('\n')
+                ;
+            }
+
+            const args = Object.keys(spec).reduce(
+                (result, k) => {
+                    let val = this.utils.removePHPObject(spec[k]);
+                    const divider = result === '' ? '' : ',\n';
+                    return `${result}${divider}    '${getKey(k)}' => ${val}`;
+                },
+                ''
+            );
+
+            return []
+                .concat(comment)
+                .concat('')
+                .concat(client)
+                .concat(collection)
+                .concat(`$cursor = $collection->find(${this.utils.removePHPObject(filter)}${args ? `, [\n${args}\n]` : ''});`)
```

Comment:
@aleksandr-rudo: Is there a syntax error here? It looks like it should be a ternary statement, but the `?` appears _within_ the backtick-quoted template literal.

Reply:
That's a copy from other language and this ternary statement works for both cases. But I'll fix it to be absolutely sure.